XRFoundation/XREngine
# Add client logging API endpoint.

* Adds a `/api/log` endpoint to server-core, which receives all log events from `client`, `engine`, and `hyperflux` and sends them to Elastic for aggregation (only when `APP_ENV !== 'development'`) 
* Fixes `server-core` log events not having a `component` property
* Adds a `multiLogger` export to `common`.
* Adds error logging to `NotificationService.dispatchNotify()` (only when `variant === 'error'`)

Usage:

```ts
import multiLogger from '@xrengine/common/src/logger'
const logger = multiLogger.child({ component: 'client-core:ClientSettingService' }) // or appropriate component name

logger.error(err)
logger.info('Message')
logger.info('Message %o', { interpolation: 'values' })
logger.info({merge: 'object'}, 'Message goes here')
logger.info({merge: 'object'}, 'Message goes here %o', { interpolation: 'values' })
```

To test sending to Elasticsearch:

Edit your `.env.local`, set `VITE_FORCE_CLIENT_LOG_AGGREGATE=true`.
(You can then see the log events in local Kibana http://localhost:5601/app/kibana )
